numba/core/typing/dictdecl.py

```diff
@@ -17,13 +17,39 @@
 
 _message_dict_support = """
 Unsupported use of `dict()` with positional or keyword argument(s). \
-The only supported use is `dict()`.
+The only supported uses are `dict()` or `dict(iterable)`.
 """.strip()
 
 
 @infer_global(dict)
 class DictBuiltin(AbstractTemplate):
     def generic(self, args, kws):
-        if args or kws:
+        if kws:
             raise errors.TypingError(_message_dict_support)
+        if args:
+            iterable, = args
+            if isinstance(iterable, types.IterableType):
+                dtype = iterable.iterator_type.yield_type
+                if isinstance(dtype, types.UniTuple):
```

Comment:
One question:

```python
@njit
def foo():
    return dict(([1, 2], [1, 2])) # Error

print(foo())
```

This is valid within CPython, But this PR only intends to add support when arguments are a list or a tuple of tuples (and not the other cases). Is that correct ?

Reply:
If possible, I'd like to leave this use case for a future PR.